src/mlpack/methods/ann/activation_functions/softplus_function.hpp

```diff
@@ -102,7 +114,16 @@ class SoftplusFunction
    */
   static double Inv(const double y)
   {
-    return y > 0 ? arma::trunc_log(arma::trunc_exp(y) - 1) : 0;
+    const double val = std::log(std::exp(y) - 1);
+    if (y > 0)
```

Comment:
This check is redundant, as the `Softplus` function will never return negative values. So the only other case is zero, which is already handled below.

Reply:
Yeah, Ur right. Since it was present from before, I thought not to change it. Now, I guess I should..